### DISCUSSION TITLE
PROBLEMAS DE CONEXÃO NA API (AXIOS)

### DISCUSSION THREAD

**Discussion by dbadaro717** [frontendbr/forum#1730] [Perguntas e Respostas]
Boa tarde!
Estou tendo problemas na conexão com a api, meu api,js está assim:

import axios from 'axios';

const api = axios.create({
  baseURL: "http://localhost:3333/",
});

export default api;

Tem aparecido os seguinte erros:
![Screenshot_20200320-143605_Expo](https://user-images.githubusercontent.com/57945667/77192192-eebc4a80-6aba-11ea-8950-92c0052c2cc7.jpg)
![Screenshot_20200320-143608_Expo](https://user-images.githubusercontent.com/57945667/77192214-f976df80-6aba-11ea-8d17-c40a782e31b7.jpg)

Alguém pode me ajudar??

**Comment by rafaelangical** [comment]
Primeiro tenta usar o ip da máquina ao invés de localhost.

Segundo faz sua chama a api com uma função async e usando try catch.

**Comment by dbadaro717** [comment]
Obrigado por vir me ajudar Rafael! 
Minha função está assim:

**Comment by dbadaro717** [comment]
async function loadDevs() {      
      const { latitude, longitude } = currentRegion;

      const response = await api.get("/search", {
        params: {
          latitude,
          longitude,
          techs
        }
      });

      setDevs(response.data.devs);
    }

**Comment by dbadaro717** [comment]
Tentei fazer a sua primeira alternativa e não funcionou.

**Comment by rafaelangical** [comment]
tenta fazer assim api.get('').then(resp => console.log(resp)).catch(err => console.log(err))

**Comment by dbadaro717** [comment]
Continua o erro.....!!!!

**Comment by rafaelangical** [comment]
manda o código completo em um gist

**Comment by dbadaro717** [comment]
OK!

**Comment by dbadaro717** [comment]
BACKEND:  https://github.com/julioaze/radar
FRONTEND: https://github.com/julioaze/mobile

**Comment by rafaelangical** [comment]
https://github.com/julioaze/mobile/blob/master/src/pages/Main.js#L37 
https://github.com/julioaze/mobile/blob/master/src/pages/Main.js#L47

usa await

**Comment by dbadaro717** [comment]
Como deveria ficar ett??

**Comment by dbadaro717** [comment]
Sou novo no React teria como você me mandar o codigo..... se não for pedir mto.

**Comment by rafaelangical** [comment]
Entra nesse slack https://react-brasil-slack.herokuapp.com/

e manda as dúvidas no channel #general que ajudamos por lá, fica mais fácil.

**Comment by thejpdev** [comment]
@dbadaro717 Cara, eu percebi que vc está utilizando HTTPS no back e HTTP no front, talvez seja isso, troca e tenta novamente, ve se corrigiu

**Comment by marcuxyz** [comment]
Pode ser alguma coisa voltando errado no backend. Na hora que você ta tentando consultar, comenta tudo e retorna um `json `vázio e tenta fazer a chamada com o [axios](url). E apenas dê um `console.log` no `res.data`

**Comment by juninhoall** [comment]
Primeiro testa no Insomnia / Postman as suas requisições.... Deve se lembrar que o a verificação do certificado SSL deve estar desabilitada, caso não haja o mesmo.
